sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -2969,11 +2969,27 @@ object functions {
   }
 
   /**
-   * (Java-specific) Parses a column containing a JSON string into a `StructType` with the
+   * (Scala-specific) Parses a column containing a JSON array string into a `ArrayType` with the
    * specified schema. Returns `null`, in the case of an unparseable string.
    *
-   * @param e a string column containing JSON data.
-   * @param schema the schema to use when parsing the json string
+   * @param e a string column containing JSON array data.
+   * @param schema the schema to use when parsing the json array string
+   * @param options options to control how the json is parsed. accepts the same options and the
+   *                json data source.
+   *
+   * @group collection_funcs
+   * @since 2.2.0
+   */
+  def from_json(e: Column, schema: ArrayType, options: Map[String, String]): Column = withExpr {
```

Comment:
I thought changing `StructType` to `DataType` breaks binary compatibility as method signature is changed and the app complied in 2.1.0 does not run in 2.2.0.